components/datetime/src/options/components.rs

```diff
@@ -489,6 +492,183 @@ impl From<TimeZoneName> for Field {
     }
 }
 
+/// Get the resolved components for a DateTimeFormat, via the PatternPlurals. In the case of
+/// plurals resolve off of the required `other` pattern.
+impl<'data> From<&PatternPlurals<'data>> for Bag {
```

Comment:
The `components::Bag` is a little weird to return here, but seems more maintainable than creating a separate `ResolvedComponents` type, since the return information fully fit into the existing struct.